sycl/include/sycl/usm.hpp

```diff
@@ -19,135 +19,269 @@ namespace sycl {
 ///
 // Explicit USM
 ///
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const device &dev, const context &ctxt,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const device &dev, const context &ctxt,
-    const property_list &propList,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const queue &q,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const queue &q, const property_list &propList,
-    const detail::code_location CodeLoc = detail::code_location::current());
+__SYCL_EXPORT void *
+malloc_device(size_t size, const device &dev, const context &ctxt
+#ifndef DISABLE_SYCL_INSTRUMENTATION_METADATA
+              ,
+              const detail::code_location &CL = detail::code_location::current()
+#endif
+);
```

Comment:
Shouldn't we remove (or deprecate) overloads w/o property_list? The spec declarations have property_list parameter.

Reply:
> Shouldn't we remove (or deprecate) overloads w/o property_list? The spec declarations have property_list parameter.

I wish we could do that, and I actually removed them initially.
But they are needed due to how FE instruments the calls, it adds the code_location after last given argument before the default initialized property_list is resolved, so I got lots of fails of unresolved symbols (where propery_list wasn't on the call).